components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -104,6 +104,13 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
         if (StringUtils.isEmpty(authenticatedUser.getUserResidentOrganization())) {
             authenticatedUser.setUserResidentOrganization(appResideOrgId);
         }
+
+        /* Remove user organization from the authenticated user when switching to the app reside organization in order
+        to preserve the associative nature of the authenticated user object */
+        if (appResideOrgId.equals(accessingOrgId) &&
```

Comment:
can appResideOrgId be null?

Reply:
No, it was retrieved from https://github.com/wso2-extensions/identity-oauth2-grant-organization-switch/pull/26/files#diff-4582f045c1766b28a97c98c77e5dfd135e802b829bb8700e4d06c24a5bb2aef9R88
Hence it can't be null